src/domain/services/import.service.ts

```diff
@@ -236,14 +236,20 @@ export class ImportService {
    * SQL is then run to insert this into the individual tag tables.
    * This was found to be quicker than using ORM functionality
    */
-  async updateTags(updateId: string, fullImport = false) {
+  async updateTags(
+    updateId: string,
+    fullImport = false,
+    source = ProductSource.FULL_LOAD,
+  ) {
+    const autoCommit = source !== ProductSource.EVENT;
```

Comment:
It feels a bit weird to me to call it autoCommit, because [auto-commit for me refers to having one transaction per query, managed directly by the database](https://www.postgresql.org/docs/current/ecpg-sql-set-autocommit.html) (whereas here you will manage the commit yourself).

I would call it perProductCommit or something along this line.